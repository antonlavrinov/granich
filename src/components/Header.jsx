import { Link } from "gatsby"
import PropTypes from "prop-types"
import React from "react"
import styled from 'styled-components';
import {Container} from './style';
import logo from '../assets/images/logo-white.png';
import arrow from '../assets/images/arrow.png';
import WhiteLogo from '../assets/svgs/granich-logo.svg';
import ArrowDown from '../assets/svgs/header-arrow-icon-down.svg';
import ArrowOut from '../assets/svgs/header-arrow-icon-out.svg';
import ArrowBack from '../assets/svgs/header-arrow-back.svg';
import LoginButtonIcon from '../assets/svgs/login-button-icon.svg';
import scrollTo from 'gatsby-plugin-smoothscroll';



const HeaderSection = styled.header`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  width: 100%;

`

const LogoWrapper = styled(props => <Link {...props} />)`
  display: flex;
  align-items: center;
  fill: white;
  ${(props) => props.propstyle && `
    fill: var(--granich-black);
    
    `}
  `

const HeaderWrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 3.2vw 0;
`

const NavLinks = styled.nav`
  color: white;
  margin-left: 3.15vw;
  margin-top: -1vw;
  ul {
    display: flex;
  }
`

const NavLink = styled(props => <Link {...props} />)`
  position: relative;
  color: white;
  font-size: 1.15vw;
  text-transform: uppercase;
  font-weight: 700;
  margin-right: 3.2vw;
  padding-bottom: 0.4vw;
  :hover {
    font-size: 1.15vw;
    font-weight: 700;
    color: white;
    cursor: pointer;
    border-bottom: solid 0.15vw white;

  }
  ${(props) => props.propstyle && `
  color: var(--granich-black);
  :hover {
    font-size: 1.2vw;
    font-weight: 700;
    color: var(--granich-black);
    cursor: pointer;
    border-bottom: solid 0.15vw var(--granich-black);
  }
`}

`
const NavLinkExternal = styled.a`
  position: relative;
  color: white;
  font-size: 1.2vw;
  text-transform: uppercase;
  font-weight: 700;
  margin-right: 3.2vw;
  padding-bottom: 0.4vw;
  :hover {
    font-size: 1.2vw;
    font-weight: 700;
    color: white;
    cursor: pointer;
    border-bottom: solid 0.15vw white;

  }
  ${(props) => props.propstyle && `
  color: var(--granich-black);
  :hover {
    font-size: 1.2vw;
    font-weight: 700;
    color: var(--granich-black);
    cursor: pointer;
    border-bottom: solid 0.15vw var(--granich-black);
  }
  `}
`

const ArrowLinkDown = styled(props => <ArrowDown {...props}/>)`
  position: absolute;
  top: -0.5vw;
  right: -1.4vw;


`

const ArrowLinkOut = styled(props => <ArrowOut {...props}/>)`
  position: absolute;
  top: -0.5vw;
  right: -1.4vw;


`
const ArrowLinkBack = styled(props => <ArrowBack {...props}/>)`
  position: absolute;
  top: -0.5vw;
  right: -1.4vw;


`

const LoginButton = styled.a`
  display: flex;
  align-items: center;
  color: white;
  background: var(--granich-red-gradient);
  margin-top: -1.6vw;
  margin-left: auto;
  padding: 0.6vw 1.1vw;
  letter-spacing: 0.05vw;
  border-radius: 0.5vw;
  font-size: 1.15vw;
  text-transform: uppercase;
  font-weight: 600;
  transition: all 0.2s ease;
  user-select: none;
  svg {
    width: 1.1vw;
    margin-right: 0.35vw;
  }
  :hover {
    transform: scale(1.1);
    cursor: pointer;
    color: white;
    font-size: 1.15vw;
    font-weight: 600;
  }
`



const Header = ({ siteTitle, style }) => (
  <HeaderSection id="header">
      <Container>
        <HeaderWrapper>

          <LogoWrapper to="/" propstyle={style}>
            <WhiteLogo  />
          </LogoWrapper>
          <NavLinks >
            <ul >
              <li>
                {style === 'dark' ? (
                    <NavLink to="/#courses" propstyle={style}>
                      Курсы
                      {style === 'dark' ? <ArrowLinkBack/> : <ArrowLinkDown/>}
                    </NavLink>
                  ) : (
                    <NavLinkExternal onClick={() => scrollTo('#courses')} propstyle={style}>
                      Курсы
                      {style === 'dark' ? <ArrowLinkBack/> : <ArrowLinkDown/>}
                    </NavLinkExternal>
                )}
              </li>  
              <li>
                {style === 'dark' ? (
                  <NavLink to="/#preparation" propstyle={style}>
                    Подготовка
                    {style === 'dark' ? <ArrowLinkBack/> : <ArrowLinkDown/>}
                  </NavLink>
                ) : (
                  <NavLinkExternal onClick={() => scrollTo('#preparation')} propstyle={style}>
                    Подготовка
                    {style === 'dark' ? <ArrowLinkBack/> : <ArrowLinkDown/>}
                  </NavLinkExternal>
                )}

              </li>  
              <li>
                <NavLinkExternal href="https://granich.ru/cms/system/login" target="_blank" propstyle={style}>
                  Блог
                  <ArrowLinkOut/>
                </NavLinkExternal>
              </li>  

            </ul>
          </NavLinks>
          <LoginButton href="https://granich.ru/cms/system/login" target="_blank">
            <LoginButtonIcon/>Войти
          </LoginButton>

        </HeaderWrapper>
      </Container>
  </HeaderSection>

)

Header.propTypes = {
  siteTitle: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
