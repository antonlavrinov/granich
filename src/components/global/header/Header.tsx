import React from "react"
import styled from "styled-components"
import Logo from "../../../assets/svgs/granich-logo.svg"
import ArrowDown from "../../../assets/svgs/header-arrow-icon-down.svg"
import ArrowOut from "../../../assets/svgs/header-arrow-icon-out.svg"
import ArrowBack from "../../../assets/svgs/header-arrow-back.svg"

import Link from "gatsby-link"

interface ITheme {
  theme: "dark" | "light"
}

export const WhiteLogo = styled(props => <Logo {...props} />)`
  width: 4.8vw;
  /* min-width: 4.8vw; */
  height: 3.1vw;
  /* min-height: 3.1vw; */
  svg {
    fill: white;
  }
  @media only screen and (max-width: 575px) {
    width: 12.5vw;
    height: 11vw;
    /* min-width: 12.5vw;
    min-height: 11vw; */
  }
`

export const StyledHeader = styled.header`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 2;
  width: 100%;
`

export const LogoWrapper = styled.a<ITheme>`
  display: flex;
  align-items: center;
  /* fill: white; */
  height: auto;
  margin-right: 3.15vw;
  /* ${props =>
    props.theme === "dark" &&
    `
    fill: var(--granich-black);
    
    `} */
  img {
    width: 3.6vw;
    min-width: 3.6vw;
    height: 3.6vw;
    min-height: 3.6vw;
  }

  @media only screen and (max-width: 575px) {
    display: none;
    img {
      width: 12vw;
      min-width: 12vw;
      height: 12vw;
      min-height: 12vw;
    }
  }
`

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 3.2vw 0;
  @media only screen and (max-width: 575px) {
    flex-direction: column;
    align-items: flex-start;
    padding: 7.5vw 6.5vw;
  }
`

export const NavLinks = styled.nav`
  color: white;
  margin-top: 0.4vw;
  ul {
    display: flex;
    align-items: center;
  }
  @media only screen and (max-width: 575px) {
    margin-left: 0vw;
    margin-top: 0vw;
    ul {
      li {
        margin-right: 11vw;
        :last-child {
          margin-right: 0;
        }
      }
    }
  }
`

export const NavLink = styled(props => <Link {...props} />)`
  display: block;
  position: relative;
  color: white;
  font-size: 1.15vw;
  text-transform: uppercase;
  font-weight: 700;
  margin-right: 3.2vw;
  padding-bottom: 0.4vw;
  border-bottom: solid 0.15vw rgba(0, 0, 0, 0);
  :hover {
    font-size: 1.15vw;
    font-weight: 700;
    color: white;
    cursor: pointer;
    border-bottom: solid 0.15vw white;
  }

  ${props =>
    props.theme === "dark" &&
    `
  color: var(--granich-black);
  :hover {
    color: var(--granich-black);
    cursor: pointer;
    border-bottom: solid 0.15vw var(--granich-black);
  }
  `}
  @media only screen and (max-width: 575px) {
    font-size: 3.8vw;
    margin-right: 0;
    padding-bottom: 1vw;
    border-bottom: solid 0.5vw rgba(0, 0, 0, 0);
    :hover {
      font-size: 3.8vw;
      border-bottom: solid 0.5vw white;
    }
    ${props =>
      props.theme === "dark" &&
      `
    :hover {
      border-bottom: solid 0.5vw var(--granich-black);
    }
    `}
  }
`
export const NavLinkExternal = styled.a<ITheme>`
  display: block;
  position: relative;
  color: white;
  font-size: 1.15vw;
  text-transform: uppercase;
  font-weight: 700;
  margin-right: 3.2vw;
  padding-bottom: 0.4vw;
  border-bottom: solid 0.15vw rgba(0, 0, 0, 0);
  :hover {
    font-size: 1.15vw;
    font-weight: 700;
    color: white;
    cursor: pointer;

    border-bottom: solid 0.15vw white;
  }
  :focus {
  }
  ${props =>
    props.theme === "dark" &&
    `
  color: var(--granich-black);
  :hover {
    font-size: 1.15vw;
    font-weight: 700;
    color: var(--granich-black);
    cursor: pointer;
    border-bottom: solid 0.15vw var(--granich-black);
  }
  `}
  @media only screen and (max-width: 575px) {
    font-size: 3.8vw;
    margin-right: 0vw;
    padding-bottom: 1vw;
    border-bottom: solid 0.5vw rgba(0, 0, 0, 0);
    :hover {
      font-size: 3.8vw;
      border-bottom: solid 0.5vw white;
    }
    ${props =>
      props.theme === "dark" &&
      `
    :hover {
      font-size: 3.8vw;
      border-bottom: solid 0.5vw var(--granich-black);
    }
    `}
  }
`

export const ArrowLinkDown = styled(props => <ArrowDown {...props} />)`
  position: absolute;
  top: -0.5vw;
  right: -1.4vw;
  width: 1vw;
  height: 1vw;
  min-width: 1vw;
  min-height: 1vw;
  @media only screen and (max-width: 575px) {
    top: -1.6vw;
    right: -3.7vw;
    width: 3.1vw;
    height: 3.1vw;
    min-width: 3.1vw;
    min-height: 3.1vw;
  }
`

export const ArrowLinkOut = styled(props => <ArrowOut {...props} />)`
  position: absolute;
  top: -0.5vw;
  right: -1.4vw;
  width: 1vw;
  height: 1vw;
  @media only screen and (max-width: 575px) {
    top: -1.6vw;
    right: -3.7vw;
    width: 3.1vw;
    height: 3.1vw;
  }
`
export const ArrowLinkBack = styled(props => <ArrowBack {...props} />)`
  position: absolute;
  top: -0.5vw;
  right: -1.4vw;
  width: 1vw;
  height: 1vw;
  @media only screen and (max-width: 575px) {
    top: -1.6vw;
    right: -3.7vw;
    width: 3.1vw;
    height: 3.1vw;
  }
`

export const LoginButton = styled.a`
  display: flex;
  align-items: center;
  color: white;
  background: linear-gradient(90deg, rgba(230,25,51,1) 0%, rgba(213,12,34,1) 50%, rgba(189,3,27,1) 100%);
  // margin-top: -1.6vw;
  margin-left: auto;
  padding: 0.6vw 1.1vw;
  letter-spacing: 0.05vw;
  border-radius: 0.5vw;
  font-size: 1.15vw;
  text-transform: uppercase;
  font-weight: 500;
  transition: all 0.2s ease;
  user-select: none;
  box-shadow: 0.25vw 0.25vw 0.4vw rgba(0, 0, 0, 0.25);
  svg {
    width: 1.1vw;
    margin-right: 0.35vw;
  }
  :hover {
    transform: scale(1.1);
    cursor: pointer;
    color: white;
    font-size: 1.15vw;
    font-weight: 500;
    letter-spacing: 0.05vw;
  }
  @media only screen and (max-width: 575px) {
    display: none;
  }
`
export const LogoMobileWrapper = styled.a<ITheme>`
  display: none;
  @media only screen and (max-width: 575px) {
    display: flex;
    align-items: center;
    fill: white;
    height: auto;
    img {
      width: 12vw;
      min-width: 12vw;
      height: 12vw;
      min-height: 12vw;
    }
    ${props =>
      props.theme === "dark" &&
      `
      fill: var(--granich-black);
      
      `}
  }
`

export const LogoImg = styled.img`

    width:3.6vw;
    min-width: 3.6vw;
    height: 3.6vw;
    min-height: 3.6vw;

`

export const LogoAndLoginButtonWrapper = styled.div`
  display: none;
  @media only screen and (max-width: 575px) {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    margin-bottom: 6vw;
  }
`

export const LoginMobileButton = styled.a`
  display: none;
  @media only screen and (max-width: 575px) {
    box-shadow: 0.7vw 0.7vw 1.5vw rgba(0, 0, 0, 0.3);
    display: flex;
    align-items: center;
    color: white;
    background: linear-gradient(90deg, rgba(230,25,51,1) 0%, rgba(213,12,34,1) 50%, rgba(189,3,27,1) 100%);
    padding: 2vw 4.8vw 2vw 3.8vw;
    letter-spacing: 0.05vw;
    border-radius: 1.5vw;
    font-size: 3.5vw;
    text-transform: uppercase;
    font-weight: 600;
    transition: all 0.2s ease;
    user-select: none;
    svg {
      width: 3.3vw;
      margin-right: 1vw;
    }
    :hover {
      cursor: pointer;
      color: white;
      font-size: 3.5vw;
      letter-spacing: 0.05vw;
      font-weight: 600;
      transform: none;
    }
  }
`
