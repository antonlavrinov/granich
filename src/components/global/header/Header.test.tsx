import React from "react"
// import renderer from "react-test-renderer"
import { shallow } from "enzyme"
import Header from "."
import { NavLinkExternal } from "./Header"

//Если у нас внутри компонента Header есть еще другие компоненты,
//то shallow их проигнорирует
//shallow полезен тогда, когда нужно протестироват один компонент

//mount полезен, когда компонент использует браузерное api
//(например querySelector) или у него есть доступ к методам жизненного цикла
//render рендерит статический компонент

//вместо того, чтобы КАЖДЫЙ компонент проверять, имеется ли у него те элементы, мы можем просто
//сделать snapshot
//enzyme сделает снапшот компонента. И если мы что-то изменили в нем, то тест провалится, потому что снапшоты
//не будут совпадать
//когда в консоли отобразится ошибка мы можем нажать w и u, чтобы enzyme сделал еще один снапшот
//в принципе нам больше и не нужно писать никаких тестов, snapshot всё делает за нас

//tests coverage npm test -- --coverage покажет, какие файлы мы не покрыли тестами
// import React from 'react'

describe("<Header> component", () => {
  describe("with Light mode", () => {
    let wrapper
    let mockScrollTo

    beforeEach(() => {
      mockScrollTo = jest.fn()
      wrapper = shallow(<Header type={"light"} scrollTo={mockScrollTo} />)
    })

    it("should render correctly", () => {
      expect(wrapper).toMatchSnapshot()
    })
    it("should call onScroll when we click 'Курсы'", () => {
      wrapper.find(NavLinkExternal).at(0).simulate("click")
      expect(mockScrollTo).toHaveBeenCalled()
    })
    it("should call onScroll when we click 'Контент'", () => {
      wrapper.find(NavLinkExternal).at(1).simulate("click")
      expect(mockScrollTo).toHaveBeenCalled()
    })
  })
  describe("with Dark mode", () => {
    let mockScrollTo = jest.fn()

    it("should render correctly", () => {
      expect(<Header type={"dark"} scrollTo={mockScrollTo} />).toMatchSnapshot()
    })
  })
})
