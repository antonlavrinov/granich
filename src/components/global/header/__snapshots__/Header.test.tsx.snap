// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> component with Dark mode should render correctly 1`] = `
<Header
  scrollTo={[MockFunction]}
  type="dark"
/>
`;

exports[`<Header> component with Light mode should render correctly 1`] = `
<styled.header
  id="header"
>
  <styled.div>
    <styled.div>
      <styled.a
        href="/"
        theme="light"
      >
        <Styled(Component) />
      </styled.a>
      <styled.div>
        <styled.a
          href="/"
          theme="light"
        >
          <Styled(Component) />
        </styled.a>
        <styled.a
          href="https://school.granich.design/cms/system/login"
          rel="noopener noreferrer"
          target="_blank"
        >
          <test-file-stub />
          Войти
        </styled.a>
      </styled.div>
      <styled.nav>
        <ul>
          <li>
            <styled.a
              onClick={[Function]}
              tabIndex={0}
              theme="light"
            >
              Курсы
              <Styled(Component) />
            </styled.a>
          </li>
          <li>
            <styled.a
              onClick={[Function]}
              tabIndex={0}
              theme="light"
            >
              Контент
              <Styled(Component) />
            </styled.a>
          </li>
          <li>
            <styled.a
              href="/blog"
              rel="noopener noreferrer"
              target="_blank"
              theme="light"
            >
              Блог
              <Styled(Component) />
            </styled.a>
          </li>
        </ul>
      </styled.nav>
      <styled.a
        href="https://school.granich.design/cms/system/login"
        rel="noopener noreferrer"
        target="_blank"
      >
        <test-file-stub />
        Войти
      </styled.a>
    </styled.div>
  </styled.div>
</styled.header>
`;
